### PR TITLE
Team6 Agent Patch 1.1

### PR DESCRIPTION
**Pull request by dejay-vu**
# Summary

- IIFO prediction system refactored
- update voting
- update foraging
- update tax contribution

## Additional Information

Auxiliary information goes here.

## Test Plan

- no tests

### CODE REVIEW

File: internal/clients/team6/client.go [hunk 1]
```diff
@@ -130,9 +130,19 @@ func (c *client) DisasterNotification(dR disasters.DisasterReport, effects disas
...
 	}
 
-	//for team, prediction := range c.receivedDisasterPredictions {
-	// TODO: increases the trust rank based on the predictions
-	//}
+	ourDiff := math.Abs(c.disasterPredictions[c.GetID()].Magnitude - disasterhappening.Magnitude)
+	theirDiff := float64(0)
+
+	for team, prediction := range c.disasterPredictions {
+		theirDiff = math.Abs(prediction.Magnitude - disasterhappening.Magnitude)
```
Comment (NGovani) [inline internal/clients/team6/client.go:137]: div by 0 check

**Review by NGovani — CHANGES_REQUESTED** [review]
LGTM, one div by 0 check

File: internal/clients/team6/client.go (hunk 1 above)
Comment (dejay-vu) [reply internal/clients/team6/client.go:137]: ohh stupid mistake mb

Comment (NGovani) [reply internal/clients/team6/client.go:137]: np!

File: internal/clients/team6/iifo.go [hunk 2]
```diff
@@ -1,134 +1,155 @@
...
-	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
 	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"gonum.org/v1/gonum/stat/distuv"
 )
 
-func (c *client) MakeDisasterPrediction() shared.DisasterPredictionInfo {
-	// similar to the baseclient implement: mean prediction
-	pastDisasters := c.disastersHistory
-	teamsOfferingTo := []shared.ClientID{}
+func (c client) checkIsStochastic() bool {
```
Comment (lhl2617) [inline internal/clients/team6/iifo.go:8]: why is this not a ptr receiver?

Comment (lhl2617) [reply internal/clients/team6/iifo.go:8]: it's fine becasuse you are not mutating c but i was just wondering

Comment (dejay-vu) [reply internal/clients/team6/iifo.go:8]: yeah just preventing me from changing some values accidently

Comment (lhl2617) [reply internal/clients/team6/iifo.go:8]: ah -- good that you know this :)

**Review by NGovani — APPROVED** [review]
LGTM, i just realised if this is still a draft go ahead and remove my review.

**Review by Ning-72 — APPROVED** [review]

**Pull request merged**